src/style.rs

```diff
@@ -134,9 +135,22 @@ impl ProgressStyle {
         self
     }
 
-    /// Adds a custom key that references a `&ProgressState` to the template
-    pub fn with_key(mut self, key: &'static str, f: fn(&ProgressState) -> String) -> ProgressStyle {
-        self.format_map.insert(key, f);
+    /// Adds a custom key that owns a [`ProgressTracker`] to the template. These trackers can be either stateful or stateless. Stateful trackers simply need to implement the `ProgressTracker` trait.
```

Comment:
I'd like to keep the first line as it was, without a period, and end there. Any further elaboration can go after an empty line. Here too, I don't think the documentation's focus should be on stateful vs stateless, but instead focus on the capability this enables.